index.js

```diff
@@ -743,6 +784,133 @@ module.exports = async function(options) {
     process.exit(0);
   }
 
+  // Run the named task, with no extra arguments, on all sites,
+  // locking against simultaneous runs and returning immediately
+  // if it has been started fewer than `guard` minutes ago
+
+  async function runScheduledTaskOnAllSites(task, guard) {
+    // Prevent dashboard from attempting to run the task or touch assets
+    // when it wakes up
+    dashboard = await spinUpDashboard({
+      argv: { _: [] },
+      modules: {
+        'apostrophe-assets': {
+          disabled: true
+        }
+      }
+    });
+    let locked = false;
+    const taskLog = dashboard.db.collection('aposTaskLog');
+    await taskLog.createIndex({
+      task: 1,
+      startedAt: -1
+    });
+    try {
+      await dashboard.locks.lock(`all-${task}`);     
+      locked = true;
+      const safe = new Date();
+      safe.setMinutes(safe.getMinutes() - guard);
+      const last = await taskLog.findOne({ 
+        task: `all-${task}`,
+        startedAt: {
+          $gte: safe
+        }
+      });
+      if (last) {
+        return;
+      }
+
+      await taskLog.insert({
+        task: `all-${task}`,
+        startedAt: new Date()
+      });
+
+      const req = dashboard.tasks.getReq();
+      const sites = await dashboard.sites.find(req, {}).toArray();
+      for (const site of sites) {
+        log(site, 'running ' + task);
+        const args = process.argv.slice(1);
+        args[args.findIndex(arg => arg === 'tasks')] = task;
+        args.push('--site=' + site._id);
+        await require('util').promisify(spawn)(process.argv[0], args, { encoding: 'utf8', stdio: 'inherit' });
+      }
+    } finally {
```

Comment:
I can't find where this would be on node.green, but for some reason I thought `finally` support was pretty recent. Maybe not, but flagging that just in case.